pymeasure/display/widgets/results_dialog.py

```diff
@@ -45,12 +45,16 @@ class ResultsDialog(QtWidgets.QFileDialog):
     :class:`ManagedWindowBase<pymeasure.display.windows.managed_window.ManagedWindowBase>` class
     """
 
-    def __init__(self, columns, x_axis=None, y_axis=None, parent=None):
+    def __init__(self, columns, x_axis=None, y_axis=None, parent=None, setup=True):
```

Comment:
I must admit that I don't completely understand the changes in this file: if I understand it correctly, the ResultsDialog is never called with `setup=False`, is it? So I would think that the changes in this file are not necessary.

Reply:
Added `DockResultsDialog` to make sure x_axis and y_axis are strings before passing to ResultsDialog